python-server/azure_test_server.py

```diff
@@ -6,10 +6,12 @@
 
 # TODO: Handle secret management + remove uses of 'AzureCliCredential()'
 SUBSCRIPTION_ID =
-USERNAME =
-PASSWORD =
+USERNAME = 
+PASSWORD = 
 NUM_VMS_TO_CREATE = 1
 
+BASE_URL=
```

Comment:
Should this be set to a value, since it's used later?

Also, let's separate the azure changes into a different PR.